### PR TITLE
🌧️ Partial server

### LINKED ISSUE #31: :information_source: User info route
# 👮‍♂️ What's needed
Create a route for the user info, display the info about an user (uuid, name, status)

# 📜 Sources
no sources

### LINKED ISSUE #32: 📜 Users route
# 👮‍♂️ What's needed
Create a route for users, this route is about to display information of the users on the domain, and if requested only the connected users.

# 📜 Sources
no sources

### LINKED ISSUE #33: 📬 Send direct message route
# 👮‍♂️ What's needed
Create a route that send a direct message from a client to another.

# 📜 Sources
No sources

### LINKED ISSUE #34: 📑 Get all message with user route
# 👮‍♂️ What's needed
Create a route to retrieve all direct message exchanged between two clients.

# 📜 Sources
No source

### LINKED ISSUE #35: 🛬 Join team route
# 👮‍♂️ What's needed
Create a route to join a specific team (subscribe to any event related)

# 📜 Sources
No sources

### LINKED ISSUE #36: 🏃‍♂️ Leave team route
# 👮‍♂️ What's needed
Create a route to leave a team (unsubscribing from all events within the targeted team)

# 📜 Sources
No sources

### LINKED ISSUE #37: 🎽 Get all teams route
# 👮‍♂️ What's needed
Create a route to fetch all existing routes or, if a parameter is provided, all team a client is in.

# 📜 Sources
No source

### LINKED ISSUE #38: 🆕 Create team route
# 👮‍♂️ What's needed
Create a route to create team with a description and a name. Should make the creator join the  team.

# 📜 Sources
No source

### LINKED ISSUE #39: 🤝 Get teams' users route
# 👮‍♂️ What's needed
Create a route to list team's users

# 📜 Sources
No source

### LINKED ISSUE #41: 🗣️ Create channel route
# 👮‍♂️ What's needed
Create a route that handle the creation of channel.

# 📜 Sources
No sources

### PR DESCRIPTION
**Pull request by axel-eck**
We are working on the server, but for now there is at least those route that are for sure working:

- #31
- #32
- #33 
- #34
- #35 
- #36 
- #37 
- #38 
- #39
- #41

--- 

This pull request also fixes:
- Fix double free on close
- Fix crash on empty bodies
- Fix UUID compare (was using the wrong library)

### CODE REVIEW

File: server/src/data/load_arrays.c [hunk 1]
```diff
@@ -29,7 +29,7 @@ void deserialize_client(roundtable_server_t *server, json_object_t *client)
         return;
     uuid = (json_string_t *) json_object_get(client, "uuid");
     username = (json_string_t *) json_object_get(client, "username");
-    COPY_UUID(new_client->uuid, uuid_from_string(uuid->value));
+    COPY_UUID(new_client->uuid, *uuid_from_string(uuid->value));
```
Comment (Marius-P1) [inline server/src/data/load_arrays.c:32]: you can use you function copy_uuid here

File: server/src/data/load_arrays.c [hunk 2]
```diff
@@ -51,7 +51,7 @@ void deserialize_subscribers(roundtable_server_t *server,
 
     for (size_t i = 0; i < subscribers->size; i++) {
         uuid = (json_string_t *) json_array_get(subscribers, i);
-        COPY_UUID(uuid_struct, uuid_from_string(uuid->value));
+        COPY_UUID(uuid_struct, *uuid_from_string(uuid->value));
```
Comment (Marius-P1) [inline server/src/data/load_arrays.c:54]: and here

File: server/src/routes/teams/channels/create.c [hunk 3]
```diff
@@ -5,3 +5,80 @@
...
+#include "server_utils.h"
+
+static bool body_is_valid(json_object_t *body)
+{
+    bool sender = json_object_has_key(body, "user_uuid");
+    bool team = json_object_has_key(body, "team_uuid");
+    bool name = json_object_has_key(body, "name");
+    bool description = json_object_has_key(body, "description");
+
+    return sender && name && description;
```
Comment (Marius-P1) [inline server/src/routes/teams/channels/create.c]:
```suggestion
    return sender && name && description && team;
```

**Review by Marius-P1 — DISMISSED** [review]
👌

**Review by Marius-P1 — APPROVED** [review]
👌

**Review by 6im0n — APPROVED** [review]

**Pull request merged**